### PR TITLE
Minor gem cleanup for chef-bin/chef-utils/chef-config

### PR DESCRIPTION
**Pull request by tas50**
Add metadata to chef-config
Update the descriptions for Chef Infra
Move dev deps to the gemfiles so we're consistent in all the gems in this repo
Just dep on rspec since rspec itself is mocks/expectations/core
Add rake dep to chef-bin
Nuke the unused .rspec file in chef-bin

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: chef-config/chef-config.gemspec [hunk 1]
```diff
@@ -8,12 +8,20 @@ Gem::Specification.new do |spec|
...
-  spec.summary       = %q{Chef's default configuration and config loading}
+  spec.summary       = %q{Chef Infra's default configuration and config loading library}
   spec.homepage      = "https://github.com/chef/chef"
   spec.license       = "Apache-2.0"
 
   spec.required_ruby_version = ">= 2.6.0"
 
+  spec.metadata = {
+    "bug_tracker_uri" => "https://github.com/chef/chef/issues",
+    "changelog_uri" => "https://github.com/chef/chef/CHANGELOG.md",
```
Comment (phiggins) [inline chef-config/chef-config.gemspec:19]:
```suggestion
    "changelog_uri" => "https://github.com/chef/chef/blob/master/CHANGELOG.md",
```

This 404s.

File: chef-config/Gemfile [hunk 2]
```diff
@@ -2,5 +2,9 @@ source "https://rubygems.org"
 
 gem "chef-utils", path: File.expand_path("../chef-utils", __dir__) if File.exist?(File.expand_path("../chef-utils", __dir__))
 
-# Specify your gem's dependencies in chef-config.gemspec
 gemspec
+
+group(:development, :test) do
+  gem "rake"
+  gem "rspec", "~> 3.2"
```
Comment (phiggins) [inline chef-config/Gemfile:9]: Does this version actually matter?

Comment (tas50) [reply chef-config/Gemfile:9]: I would assume at some point we needed 3.2 and we want to make sure we don't get 4.x? 🤷

File: chef-config/chef-config.gemspec (hunk 1 above)
Comment (tas50) [reply chef-config/chef-config.gemspec]: That was a copy/paste from chef-utils. I fixed it there as well.

**Review by phiggins — APPROVED** [review]

**Pull request merged**